### DISCUSSION TITLE
新增Agent pipeline Runtime，支持LLM Agent应用开发。

### DISCUSSION THREAD

**Discussion by ully** [flow-engine/.github#6] [Ideas]
新增Agent pipeline Runtime，支持RAG与AGENT应用开发。
- 开发态风格保持和其他pipeline一致
- 运行调试参考当前chatbot模式。
- 能够支持多种节点混排。
- 支持RAG和Agent应用以一致的方式开发。

详细设计内容可以内部沟通。

**Comment by adai-5090** [comment]
对上设计特点的进一步理解，如有偏差，请指出：
1.  agent flow的自动运行和自我调试修复功能——可借鉴主流的框架和应用模式，如taskweaver、Xagent、devin；与人交互来保证flow稳定性，可借鉴gpts（coze）、chatdev（开源框架，面壁智能）模式；
2.  多种节点混排——多种agent节点可混合协作构建flow，类langgraph开发模式；
3.  RAG和Agent应用一致方式——RAG本身可以作为一类agent应用参与flow，或将RAG构造为多agent的flow；

**Reply by ully** [reply]
没错，如何将它们融合在一起，让用户无感完成大模型应用的编排是关键价值。可以加我微信私聊。
